Src/ILGPU/IR/BasicBlockMapping.cs

```diff
@@ -243,6 +243,26 @@ public bool Remove(BasicBlock block)
             return removed;
         }
 
+        /// <summary>
+        /// Asserts that the set has been cleared.
+        /// </summary>
+        [SuppressMessage(
+            "Performance",
+            "CA1822:Mark members as static",
+            Justification = "For debugging purposes only")]
+        [Conditional("DEBUG")]
+        readonly partial void AssertCleared();
+
+        /// <summary>
+        /// Removes all elements from this set.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Clear()
+        {
+            Array.Clear(visited, 0, visited.Length);
```

Comment:
QUESTION: Should the `visited` array be reset?

Reply:
Actually, my mistake. Looks like I mistook the code as clearing `values` rather than `visisted`. All good.